src/controllers/adminController.ts

```diff
@@ -22,13 +22,30 @@ export const getUsers = async (req: Request, res: Response) => {
 export const updateUserRole = async (req: Request, res: Response) => {
   try {
     const { userId, newRoles } = req.body;
-    if (!userId || !newRoles) {
+
+    if (!userId || !newRoles || !Array.isArray(newRoles)) {
```

Comment:
this check will not pass if there is userId and empty roles say roles = []

Reply:
True, noted..